docker/cdash.docker

```diff
@@ -0,0 +1,117 @@
+FROM php:7.1-apache
+LABEL maintainer="Kitware, Inc. <cdash@public.kitware.com>"
+
+ARG CDASH_DB_HOST=localhost
+
+# Designates as dev build, adds testing infrastructure, et al.
+ARG DEVELOPMENT_BUILD
+
+# make sure it's set in the ENV for reference in docker-entrypoint.sh
+# TODO: rename CDASH_DEVELOPMENT_BUILD ?
+ENV DEVELOPMENT_BUILD=$DEVELOPMENT_BUILD
+
+# NODE_ENV expands to nothing if DEVELOPMENT_BUILD is unset or empty
+ENV NODE_ENV=${DEVELOPMENT_BUILD:+'development'}
+
+# NODE_ENV expands to production if NODE_ENV is unset or empty,
+# i.e. if DEVELOMPMENT_BUILD is empty the node environment is production
+ENV NODE_ENV=${NODE_ENV:-'production'}
+
+# PHP_ENV expands to empty if DEVELOPMENT_BUILD is unset or empty
+ENV PHP_ENV=${DEVELOPMENT_BUILD:+' '}
+
+# PHP_ENV expands to --no-dev (composer flag) if PHP_ENV is empty,
+# i.e. composer will not try to load require-dev packages, e.g. those
+# requiring xdebug
+ENV PHP_ENV=${PHP_ENV:-'--no-dev'}
+
+RUN apt-get update                                                         \
+ && apt-get install -y gnupg                                               \
+ && curl -sL https://deb.nodesource.com/setup_6.x | bash                   \
+ && apt-get install -y git libbz2-dev libfreetype6-dev libjpeg62-turbo-dev \
+    libmcrypt-dev libpng-dev libpq-dev libxslt-dev libxss1 nodejs unzip    \
+    wget zip                                                               \
+ && docker-php-ext-configure pgsql --with-pgsql=/usr/local/pgsql           \
+ && docker-php-ext-configure gd --with-freetype-dir=/usr/include/          \
+                                --with-jpeg-dir=/usr/include/              \
+ && docker-php-ext-install -j$(nproc) bcmath bz2 gd                        \
+    pdo_mysql pdo_pgsql xsl                                                \
+ && wget -q -O checksum https://composer.github.io/installer.sha384sum     \
+ && wget -q -O composer-setup.php https://getcomposer.org/installer        \
+ && sha384sum -c checksum                                                  \
+ && php composer-setup.php                                                 \
+    --install-dir=/usr/local/bin --filename=composer                       \
+ && php -r "unlink('composer-setup.php');"                                 \
+ && composer self-update --no-interaction
+
+# Creates the kitware home directory with directory for cdash
+WORKDIR /home/kitware/cdash
+
+# Copy CDash (current folder) into /home/kitware/cdash to set its owner
+# to www-data
+ADD . ./
```

Comment:
It is a Dockerfile [best practice](https://docs.docker.com/v17.09/engine/userguide/eng-image/dockerfile_best-practices/#add-or-copy) to prefer `COPY` over `ADD`.